jgrapht-core/src/main/java/org/jgrapht/graph/interval/IntervalGraphMapping.java

```diff
@@ -0,0 +1,354 @@
+     * stores state whether this mapping is live (that is, is a ListenableGraph used)
+     */
+    private boolean mappingLive;
+
+    /**
+     * Basic Constructor for a non-live mapping to an interval graph. All methods are passed through to the underlying interval graph.
+     *
+     * @param graph the graph to be mapped
+     */
+    private IntervalGraphMapping(Graph<V, E> graph) {
+        this.graph = graph;
+        this.intervalStructure = new IntervalTreeStructure<>();
+        this.intervalMap = new HashMap<>();
+        // as we know that a this is a correctly initialized mapping, we can set this mapping to valid
+        this.mappingValid = true;
+        // as we know that no ListenableGraph is used, we set the mapping non-live
+        this.mappingLive = false;
+    }
+
+    /**
+     * Basic constructor for a live mapping to an interval graph. All methods are passed through to the underlying interval graph.
+     * Every vertex and edge update will be considered.
+     *
+     * @param graph the graph to be mapped
+     */
+    private IntervalGraphMapping(ListenableGraph<V, E> graph) {
+        this((Graph<V, E>) graph);
+        // add the basic IntervalGraphListener to the ListenableGraph such that we have a live mapping
+        graph.addGraphListener(new IntervalGraphMappingListener<>(this));
+        // as we know that a ListenableGraph is used, we set the mapping live
+        this.mappingLive = true;
+    }
+
+    /**
+     * Constructs a new live interval graph mapping with a new graph. The graph can either be directed or undirected, depending on the
+     * specified edge supplier.
+     *
+     * @param vertexSupplier the vertex supplier of the new graph
+     * @param edgeSupplier the edge supplier of the new graph.
+     * @param weighted whether the graph is weighted, i.e. the edges support a weight attribute
+     */
+    public IntervalGraphMapping(Supplier<V> vertexSupplier, Supplier<E> edgeSupplier, boolean weighted) {
+        this(new DefaultListenableGraph<>(new DefaultUndirectedGraph<>(vertexSupplier, edgeSupplier, weighted)));
+    }
+
+    /**
+     * Construct a new live interval graph mapping with a new graph with given <code>vertices</code>. The graph can either be directed or undirected, depending on the
+     * specified edge supplier.
+     * Runs in linear time if vertex list is sorted by the starting point and the end point of the interval, otherwise naive insertion is performed
+     *
+     * @param vertices initial vertices
+     * @param vertexSupplier the vertex supplier of the new graph.
+     * @param edgeSupplier the edge supplier of the new graph.
+     * @param weighted whether the graph is weighted, i.e. the edges support a weight attribute
+     *
+     * @throws NullPointerException if the specified edge factory is <code>
+     * null</code>.
+     */
+    public IntervalGraphMapping(ArrayList<V> vertices, Supplier<V> vertexSupplier, Supplier<E> edgeSupplier, boolean weighted) {
```

Comment:
Why do you restrict the List to ArrayList? Guess you should allow every List.

Reply:
At the moment I need arbitrary access to indices... So I need this for efficiency reasons. Unfortunately, the java library does not consider efficiency in their modules and interfaces, so I circumvent it by that...
Maybe I will change that while implementing the edge insertion more efficiently.

Reply:
fixed